cargo-dist/src/backend/installer/homebrew.rs

```diff
@@ -49,6 +51,40 @@ pub(crate) fn write_homebrew_formula(
 ) -> DistResult<()> {
     let mut info = source_info.clone();
 
+    // Collect all dist-manifests and fetch the appropriate Mac ones
+    let mut manifests = vec![];
+    for file in glob(&format!("{}/*-dist-manifest.json", &graph.dist_dir))? {
+        let json_path = Utf8PathBuf::try_from(file?)?;
+        let data = SourceFile::load_local(json_path)?;
+        let manifest: DistManifest = data.deserialize_json()?;
+
+        if manifest.linkage.clone().into_iter().any(|l| {
+            info.arm64.is_some() && l.target == "aarch64-apple-darwin"
+                || info.x86_64.is_some() && l.target == "x86_64-apple-darwin"
+        }) {
+            manifests.push(manifest);
+        }
+    }
+
+    // Fetch any detected dependencies from the linkage data
+    let mut dependencies: Vec<String> = manifests
+        .into_iter()
+        .flat_map(|m| {
+            m.linkage
+                .into_iter()
+                .map(|l| {
+                    l.homebrew
+                        .into_iter()
+                        .filter_map(|lib| lib.source)
+                        .collect()
+                })
+                .collect::<Vec<String>>()
+        })
+        .collect();
+
+    // Merge with the manually-specified deps
+    info.dependencies.append(&mut dependencies);
```

Comment:
i believe all 3 of these collects can be removed if you just replace `append` with `extend`.

Reply:
Looks like we still need `l.homebrew`'s `collect` for it to figure out that the type has changed to `String` instead of `Library`, but otherwise, yeah - simplifies it. I'll push that up.